api/WebSocket.json

```
@@ -82,47 +71,40 @@
           "description": "<code>WebSocket()</code> constructor",
           "support": {
             "chrome": {
-              "version_added": true
+              "version_added": "4"
             },
             "chrome_android": {
-              "version_added": true
+              "version_added": "18"
             },
             "edge": {
               "version_added": "12"
             },
-            "firefox": [
-              {
-                "version_added": "7"
-              },
-              {
-                "version_added": "4",
-                "version_removed": "7",
-                "notes": "Parameter <code>protocols</code> not supported."
-              }
-            ],
+            "firefox": {
+              "version_added": "7"
+            },
             "firefox_android": {
               "version_added": "7"
             },
             "ie": {
               "version_added": "10"
             },
             "opera": {
-              "version_added": true
+              "version_added": "≤15"
```

The data for constructors should usually match the parent feature, especially for things where the constructor is the main entry point to the API.

The parent feature says 12.1 and https://dev.opera.com/blog/opera-12-10-is-out/ mentions "Web Sockets" so this should also be 12.1:

```suggestion
              "version_added": "12.1"
```